### ISSUE TITLE
Every form is marked as spam (Submission marked as spam - “Failed Captcha “Recaptcha”: “Client-side token missing.”) after migration to Craft 4

### ISSUE THREAD

**Issue by Antimated** [verbb/formie#1600]
### Describe the bug

Hello!

I recently migrated my Craft 3.9.3 install to Craft 4.x. Formie upgrade went from version 1.6.33 to version 2.0.37.
Before the migration my reCAPTCHA V3 integration worked fine. Since the upgrade I keep getting the following message in my log files:

```
2023-10-17 16:55:12 [INFO] Submission triggered for contact.
2023-10-17 16:55:12 [INFO] Submission marked as spam - “Failed Captcha “Recaptcha”: “Client-side token missing.”” - {"siteLanguage":"en","yourFullName":"x","yourEmailAddress":"test@mydomain.com","yourPhoneNumber":"{\"number\":\"xxx\",\"country\":null,\"hasCountryCode\":false}","yourMessage":"xxx","company":"xxxcompany","address":"{\"autocomplete\":null,\"address1\":\"xxx\",\"address2\":null,\"address3\":null,\"city\":\"xxx\",\"state\":\"xxx\",\"zip\":\"6666\",\"country\":\"BE\"}"}.
```

I double checked my reCAPTCHA keys and version on google's reCAPTCHA page and in the Formie settings and they are identical.

Here is a screenshot of my reCAPTCHA integration settings (keys excluded)

![Screenshot 2023-10-17 at 17 06 46](https://github.com/verbb/formie/assets/2869409/568a9046-8a28-4c4b-a2fc-fbcf720d9a27)

I'm not sure what is going wrong here. Hoping to get a helping hand!

If there is any more info you need that the info given, please let me know

P.S.: The form I tested on was a multi-page form, but the issue also occurs on single page forms.

### Steps to reproduce

1. Add reCAPTCHA integration and enable it
2. Enable reCAPTCHA integration on the form itself
3. Try and submit a form and check log files

### Form settings

- Multi-page form: Tested both, both break
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: Yes (I disabled the custom templates and the issue persists)

### Craft CMS version

Craft Pro 4.5.6.1

### Plugin version

2.0.38

### Multi-site?

Yes

### Additional context

Happens on local and production environment. Both domains are whitelisted in google reCAPTCHA.

**Comment by engram-design** [comment]
So this error points to the fact that the `g-recaptcha-response` param is missing from the payload sent with the form submission. You can confirm this by checking your Network tools (in Chrome Developer Tools). For example:

![image](https://github.com/verbb/formie/assets/1221575/7ea8c103-7d41-41a8-bd83-af43ed6221cd)

Are there any JS errors when submitting the form? How are you rendering the form? Is Formie's JS being included all okay?

**Comment by Antimated** [comment]
@engram-design 

Indeed I do not find the `g-recaptcha-response` parameter in the payload.
I also get the following error after a little while (not when submitting though).

<img width="1637" alt="Screenshot 2023-10-18 at 08 37 46" src="https://github.com/verbb/formie/assets/2869409/b5cdd846-2b94-461a-a862-e3694ffd37d5">

I think Formie's js is being included okay, here is a screenshot what is being output just before the closing body tag:

<img width="1342" alt="Screenshot 2023-10-18 at 09 01 11" src="https://github.com/verbb/formie/assets/2869409/315671d1-4885-4882-a0be-5831038cc962">

If I copy the the code from the `FORMIE_RECAPTCHA_SCRIPT` script (just as a test) and manually add the javascript at the end of the body tag it displays a recapatcha.

Here is an example of how the form block displays a form:

![Screenshot 2023-10-18 at 08 52 59](https://github.com/verbb/formie/assets/2869409/1a2c58d6-edb0-4c0d-801a-84f4e73a1819)


These are my form template settings:
![Screenshot 2023-10-18 at 08 50 50](https://github.com/verbb/formie/assets/2869409/7cb4e050-5e6f-426c-b61b-abeca81a1f12)

**Comment by engram-design** [comment]
Hmm, so seems like reCAPTCHA is failing to load, as you can see from the timeout. Can you rule out any network issues on your end with reaching their server? Is the captcha actually rendering?

It also goes without saying that you've checked out the [upgrade guide](https://verbb.io/craft-plugins/formie/docs/get-started/upgrading-from-v1#form-id) particularly any changes you might be making to the Form ID.

**Comment by Antimated** [comment]
> Hmm, so seems like reCAPTCHA is failing to load, as you can see from the timeout. Can you rule out any network issues on your end with reaching their server? Is the captcha actually rendering?

No network issues as it happens locally, on staging and production. The captcha is not rendering. The only instance where the reCAPTCHA renders is if I copy the code that is inside the `FORMIE_RECAPTCHA_SCRIPT` script tag and paste that directly before my closing body tag (inside a script tag ofcourse).

The only captcha related thing that is rendering in the form is inside the form: `<div data-recaptcha-placeholder="" class="formie-recaptcha-placeholder"></div>`. The captcha is set to a "Bottom-right" badge type so I'm not sure why this empty div is rendering inside the form.

> It also goes without saying that you've checked out the [upgrade guide](https://verbb.io/craft-plugins/formie/docs/get-started/upgrading-from-v1#form-id) particularly any changes you might be making to the Form ID.

I'm not modifying the the `form-id` at all. The fetched form is just passed in

```twig

                 {% set form = self.blockForm.one() %}

                  {{ craft.formie.renderForm(form, {
                      themeConfig: {
                          backButton: {
                              attributes: {
                                  class: ['button','button--primary']
                              }
                          },
                          saveButton: {
                              attributes: {
                                  class: ['button','button--primary']
                              }
                          },
                          submitButton: {
                              attributes: {
                                  class: ['button','button--primary']
                              }
                          }
                      }
                  }) }}
```

**Comment by engram-design** [comment]
The placeholder rendering is correct, that just helps to know when to load the captcha, particularly for multi-page forms. It's also for something we're about to implement that when that placeholder is "visible" the captcha scripts are lazy-loaded in. But that's not implemented just yet.

Any chance you could share your staging URL to `support@verbb.io` so I can take a look at what's going on?

**Comment by engram-design** [comment]
Thanks for sending that through. I can see there's a warning in the JS console:

```
Formie scripts already loaded for form #fui-contact1-aycvgx
```

Which means that something is trying to load scripts more than once. I can see in your `dist/js/index.2a8280615665e10ad142.js` JS file that you're calling `window.Formie.initForms()` - is this correct? You shouldn't need to initialise the forms, as they are automatically done on-load. But I can see you're using Vue, so you're likely re-initializing them?

We did add some extra handling to protect against double-binding the JS to the form, as people were mistakenly doing this. But for Vue, this is kind of required, due to the VDOM taking over, and Formie's JS will only initialize things for the real DOM, before Vue takes over.

I've improved this to handle re-binding things a bit better. Previously, this wouldn't work when used with Vue in some circumstances (a race condition). To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.38"`.

**Comment by Antimated** [comment]
I indeed call `window.Formie.initForms()` because I use Vue and have to rebind the Formie listeners. I installed the latest version you mentioned but still the issue persists :( 

I'll push a new build and drop it on the staging server (having some issues with bitbucket now)

**Comment by engram-design** [comment]
Hmm, an updated staging server would be handy if you can! I'm not quite sure what's going on now...

**Comment by Antimated** [comment]
One final note:
Seems like the data-fui-scripts `div` now is added multiple times (2nd time after `window.Formie.initForms();` is running)

![Screenshot 2023-10-19 at 14 34 34](https://github.com/verbb/formie/assets/2869409/905ed317-df30-4a79-a490-f58803e9aa67)

**Comment by engram-design** [comment]
Those scripts loading twice is correct behavior, that’s the only instance where it’s okay for that to happen, as everything needs to be re-bound again to Vue’s DOM which has taken over.

**Comment by Antimated** [comment]
@engram-design Updated my staging environment :)

**Comment by engram-design** [comment]
So the issue at the moment is there's double-binding of a lot of things, and while Formie's JS does try and handle it gracefully, some things like reCAPTCHA don't like to be initialised twice. As I've mentioned, the issue here Is that reCAPTCHA is initialized on the real DOM, then Vue kicks in, and triggers Formie to initialize the forms, and tries to re-initialize reCAPTCHA on the "sort of" same element.

So for starters, I would **highly** recommend changing your render function to not automatically initialize Formie's JS. This way, you can control when you want to initialize it.

```
{{ craft.formie.renderForm('contactForm', {
    initJs: false,
}) }}
```

And you would continue using `Formie.initForms();` in your Vue app. Which could also be done in non-Vue:

```js
document.addEventListener('DOMContentLoaded', (event) => {
    Formie.initForms();
});
```

This will be the recommended approach going forward when used with Vue, or if you'd like to control rendering a little more.

However, I've also made improvements on how captchas act when re-initializing them - just in case. So things _should_ work correctly by just applying this patch, but as I've mentioned - probably a good idea to use `initJs: false` so you can save a whole lot of JS being triggered twice for no reason.

To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.38"`.

**Comment by engram-design** [comment]
Fixed in [2.0.39](https://github.com/verbb/formie/releases/tag/2.0.39)
